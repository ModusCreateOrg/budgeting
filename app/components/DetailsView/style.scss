@import 'theme/variables';

@mixin btn-theme($btn-color) {
  background-color: $btn-color;
  &:hover {
    background-color: darken($btn-color, 6%);
  }
}

h2.neg {
  color: $red;
}

h2.pos {
  color: $green;
}

.btn {
  display: flex;
  align-items: center;
  justify-content: center;
  color: #fff;
  margin-left: 5px;
  font: 400 11px system-ui;
  text-transform: uppercase;
  text-shadow: 0 1px 1px rgba(0, 0, 0, 0.2);
  border-radius: 4px;
  padding: 0.5em 1em;
  text-decoration: none;
  border: transparent;
  cursor: pointer;
  height: 35px;

  &.btnDefault {
    @include btn-theme($mediumgray);
  }
}

.details {
  display: flex;
  flex-direction: column;
  align-items: stretch;
  justify-content: flex-start;
  width: 100%;
}

.details__header {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: flex-end;
  margin-bottom: 2em;
  border: 1px solid $gray;
  border-radius: 6px;
  width: 100%;
}
