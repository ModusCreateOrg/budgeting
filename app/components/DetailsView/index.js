import React from 'react';
import formatAmount from 'utils/formatAmount';
import DonutChart from 'components/DonutChart';
import style from './style.scss';

const DetailsView = ({ transaction, category, goBack, totalIn, totalOut }) => {
  const description = (transaction && transaction.description) || '';
  const value = (transaction && transaction.value) || 0;
  const totalValue = value > 0 ? totalIn : totalOut;

  const formattedAmount = formatAmount(value).text;
  const valueAbs = Math.abs(value);
  const totalAbs = Math.abs(totalValue);

  const chartData = [
    {
      category: category,
      categoryId: 1,
      value: valueAbs,
    },
    {
      category: 'Other',
      categoryId: 2,
      value: totalAbs,
    },
  ];

  return (
    <div className={style.details}>
      <div className={style.details__header}>
        <button type="button" className={`${style.btn} ${style.btnDefault}`} onClick={goBack}>
          Go back to list
        </button>
      </div>
      {transaction && (
        <React.Fragment>
          <h1>Description: {description}</h1>
          <h2 className={transaction.value > 0 ? style.pos : style.neg}>Value: {formattedAmount}</h2>
          <br />
          <h5>Category: {category}</h5>
          <DonutChart data={chartData} dataLabel="category" dataKey="categoryId" innerRatio={999} useFormat={false} />
        </React.Fragment>
      )}
    </div>
  );
};

export default DetailsView;
