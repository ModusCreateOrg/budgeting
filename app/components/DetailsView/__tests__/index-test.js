import React from 'react';
import renderer from 'react-test-renderer';
import DetailsView from '..';

it('renders correctly', () => {
  const tree = renderer.create(<DetailsView />).toJSON();
  expect(tree).toMatchSnapshot();
});
