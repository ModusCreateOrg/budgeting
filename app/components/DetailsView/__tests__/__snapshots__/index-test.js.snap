// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="details"
>
  <div
    className="details__header"
  >
    <button
      className="btn btnDefault"
      type="button"
    >
      Go back to list
    </button>
  </div>
</div>
`;
